.github/workflows/test_cra.yml

```diff
@@ -0,0 +1,39 @@
+on:
+  pull_request:
+    types: [opened, reopened, ready_for_review, review_requested]
+  issue_comment:
+    types: [created]
+
+jobs:
+  code_review_job:
+    runs-on: ubuntu-latest
+    permissions:
+      issues: write
+      pull-requests: write
+      contents: write
+    name: Run code review agent on every pull request, respond to user comments
+
+    steps:
+      - name: Set Options Environment Variable for Issue Comment
+        if: github.event_name == 'issue_comment'
+        run: |
+          echo "OPTIONS=--static_analysis.fb_infer.enabled=True --code_feedback=True --dependency_check.enabled=False --bee.path=/automation-platform  --bee.actn_dir=/automation-platform/default_bito_ad/bito_modules --git.access_token=${{ secrets.GIT_ACCESS_TOKEN }} --bito_cli.bito.access_key=${{ secrets.BITO_ACCESS_KEY }}${{ secrets.GIT_DOMAIN && ' --git.domain=' || '' }}${{ secrets.GIT_DOMAIN }}" >> $GITHUB_ENV
+      - name: Code Review Agent - Issue Comment
+        if: github.event_name == 'issue_comment'
+        uses: gitbito/codereviewagent@BITO_QA_TEST
+        with:
+          pr: ${{ github.event.issue.pull_request.html_url }}
+          command: ${{ github.event.comment.body }}
+          options: ${{ env.OPTIONS }}
+
+      - name: Set Options Environment Variable for Pull Request
+        if: github.event_name == 'pull_request'
+        run: |
+          echo "OPTIONS=--static_analysis.fb_infer.enabled=True --code_feedback=True --dependency_check.enabled=False --bee.path=/automation-platform  --bee.actn_dir=/automation-platform/default_bito_ad/bito_modules --git.access_token=${{ secrets.GIT_ACCESS_TOKEN }} --bito_cli.bito.access_key=${{ secrets.BITO_ACCESS_KEY }}${{ secrets.GIT_DOMAIN && ' --git.domain=' || '' }}${{ secrets.GIT_DOMAIN }}" >> $GITHUB_ENV
+      - name: Code Review Agent action step
+        if: github.event_name == 'pull_request'
+        uses: gitbito/codereviewagent@BITO_QA_TEST
+        with:
+          pr: ${{ github.event.pull_request.html_url }}
+          command: review
+          options: ${{ env.OPTIONS }}
```

Comment:
**Scalability Issue**: The GitHub Actions workflow is configured to run on every pull request and issue comment without any rate limiting or conditions to prevent abuse. This could lead to resource exhaustion, especially if the actions perform heavy tasks. <br> **Fix**: Implement conditions to limit when the workflow runs, such as only on pull requests to specific branches or when specific labels are applied. Additionally, consider adding a manual trigger option. <br> **Code Suggestion**: 
 ```
 +on:
 +  pull_request:
 +    types: [opened, reopened, synchronized]
 +    branches:
 +      - main
 +  issue_comment:
 +    types: [created]
 +    branches:
 +      - main
 ```